source/includes/compatibility/mongodb-compatibility-table-c.rst

```diff
@@ -0,0 +1,401 @@
+.. sharedinclude:: dbx/compatibility-table-legend.rst
+  
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :class: compatibility-large
+
+   * - C Driver Version
+     - MongoDB 8.0
+     - MongoDB 7.0
+     - MongoDB 6.0
+     - MongoDB 5.0
+     - MongoDB 4.4
+     - MongoDB 4.2
+     - MongoDB 4.0
+     - MongoDB 3.6
+     - MongoDB 3.4
+     - MongoDB 3.2
+     - MongoDB 3.0
+     - MongoDB 2.6
```

Comment:
S: ask C DBX team if we can remove both old MDB versions and old C driver versions to shorten this table

Reply:
@kevinAlbs What do you think of removing these older MDB versions from the table? It would be similar to the [compatibility table for PyMongo](https://www.mongodb.com/docs/languages/python/pymongo-driver/current/compatibility/) which was recently updated.

Reply:
I support removing old MDB versions and old C driver versions. I expect users wanting to find compatibility for older versions could still find the information on older docs pages (https://www.mongodb.com/docs/languages/c/c-driver/v1.27/)

Reply:
Once the standardization for C driver is complete, the changes will be backported to 1.26. So by default, 1.26 and 1.27 wouldn't include the compatibility for older versions unless we specifically include them on those branches. Would including them in older docs be your team's preference? 

In the meantime, I can update this PR to include the change for you to see how it will look and review.

Reply:
> Would including them in older docs be your team's preference?

Preserving the info for older versions is my preference. But if it is extra trouble to preserve the compatibility with older versions, I expect it is low value and I think it is OK to leave out, even on older branches.